### ISSUE TITLE
am health leaks a ~41MB probe-copy directory per killed invocation (hardcoded /tmp path, cleanup only on clean exit)

### ISSUE THREAD

**Issue by genx7up** [Dicklesworthstone/mcp_agent_mail_rust#308]
## Summary

`am health` (`am robot health`) stages its integrity probe as a full copy of the mailbox database under a **hardcoded** `/tmp/.mcp-agent-mail-health-probe-<rand>/` directory (`TMPDIR` is not consulted for this path, unlike other parts of the CLI) and removes that directory only when the process exits normally. A caller that kills the `am health` invocation mid-probe — a supervisor/dashboard/poller on a tight tick is the realistic trigger — leaves the full copy behind. Each leaked copy is the size of the live mailbox DB (~41 MB on my test mailbox); a poller that health-checks every few seconds turns this into unbounded `/tmp` growth.

This looks like the same staged-copy mechanism discussed in #300 (`health-probe.sqlite3`, `health-probe.sqlite3-wal`, etc. — matching file names), whose fix addressed the probe's *failure classification* but explicitly left "retaining a bounded diagnostic artifact for the failed probe" as a disk-retention question for a follow-up issue. This report is that follow-up, for the unconditional (not just on-failure) leak.

## Reproduction (am 0.3.32, current latest)

```bash
# baseline: no leaked probe dirs
ls -1d /tmp/.mcp-agent-mail-health-probe-* 2>/dev/null | wc -l   # 0

# a normal, completed run cleans up after itself
timeout 10 am health >/dev/null 2>&1
ls -1d /tmp/.mcp-agent-mail-health-probe-* 2>/dev/null | wc -l   # still 0

# a run that gets SIGKILL'd mid-probe leaves its copy behind
timeout -s KILL 1 am health >/dev/null 2>&1
ls -1d /tmp/.mcp-agent-mail-health-probe-* 2>/dev/null           # a new dir appears
du -sh /tmp/.mcp-agent-mail-health-probe-*                       # ~41M

# reproducible on demand: 3/3 repeats each leave one more directory
for i in 1 2 3; do timeout -s KILL 1 am health >/dev/null 2>&1; done
ls -1d /tmp/.mcp-agent-mail-health-probe-* | wc -l               # +3
```

Directory contents on a leak:

```
/tmp/.mcp-agent-mail-health-probe-<rand>/
  health-probe.sqlite3
  health-probe.sqlite3-wal
  health-probe.sqlite3-shm
  health-probe.sqlite3-wal-cert
  health-probe.sqlite3-wal-cert-head
  health-probe.sqlite3-fsqlite-ns-gate
  health-probe.sqlite3-fsqlite-ns-use
```

I also tried setting `TMPDIR` to a scratch directory before the killed runs; the leaked copy did not appear there (nor consistently in `/tmp` on those particular attempts) — inconclusive on its own, but at minimum the *default*, `TMPDIR`-unset path reliably leaks in real `/tmp`, which is what an unconfigured supervisor/health-poller will hit.

## Expected

Either honor `$TMPDIR` for this probe copy (consistent with the rest of the CLI) and/or make the probe directory's lifetime tied to the process rather than to clean exit — e.g. a `tempfile`-crate-style `TempDir` guard that removes itself on `Drop` (which unwinds on most signals a supervisor would actually use, and at minimum covers the ordinary "caller times out and kills" case), or a parent-side reaper that sweeps orphaned `*-health-probe-*` directories older than a short bound. Anything that doesn't require the probed process to reach its own exit path.

## Impact

A caller with a short per-call timeout — any health-checking supervisor, TUI status bar, or dashboard polling `am health` on a fixed interval shorter than the probe's occasional worst-case latency — leaks one full DB copy per timed-out call. On a busy mailbox this is continuous, unbounded `/tmp` (or tmpfs) growth with no self-correction.

## Environment

- `am` (mcp-agent-mail) 0.3.32, `x86_64-unknown-linux-gnu`, current latest release as of this report (checksum-verified against the published `.sha256`)
- Linux 6.x, `/tmp` is a regular (non-tmpfs-constrained in this repro, but the leak is on the DB-sized copy regardless of backing fs)
- Not a duplicate of #300 — that issue and its fix are about `am doctor check` mis-classifying a *failed* staged probe as live corruption; this report is about the staged probe copy's on-disk lifetime when the *invoking process* is killed, which #300's own closing comment says was deliberately left out of scope ("the artifact would mostly be a disk-retention policy question; if you hit a case ... open a new issue").

Happy to share the exact repro script if useful.

**Comment by Dicklesworthstone** [comment]
Confirmed on the substance, fixed in 67e81417 — with one correction to the diagnosis.

What we verified:

- The leak on `SIGKILL` is real. The staging directory is a `tempfile` guard, so it is removed on a clean exit, on every early return, and on an unwinding panic — but a killed process never runs `Drop`, and nothing else ever reclaimed the copy. Each orphan is the size of the mailbox family, exactly as you measured.
- The `/tmp` path is **not** hardcoded, and `TMPDIR` was already honored: staging resolves the first non-empty `TMPDIR` / `TEMP` / `TMP` and only otherwise falls back to `std::env::temp_dir()`, which is `/tmp` on Unix when none is set. That matches your inconclusive `TMPDIR` observation — the copies landed under your scratch directory, not in `/tmp`. The resolution is now a named `snapshot_temp_root()` shared by the staging and the sweep, with a test pinning that staging lands under it.

What changed:

- Staged probe directories carry the PID of the process that created them: `.mcp-agent-mail-health-probe-<pid>-<random>`.
- The first probe staging in a process sweeps the staging root for orphans. An entry is reclaimed only if all of these hold: the name carries the probe's own prefix plus a suffix; it is a real directory, never a symlink (a link planted in a shared temp root cannot redirect the removal); it contains nothing but the probe's own staged files (`health-probe.sqlite3*`); it is not owned by this process; and its owning PID is provably gone — or, when that cannot be established (no PID segment, or a PID some live process now holds), it has not been modified for an hour. Errors are ignored throughout: reclaiming disk must never fail a health check.

The PID gate is what actually bounds your scenario. An age bound alone would still let a poller on a few-second tick accumulate an hour's worth of full-size copies before the first one aged out; instead the very next `am health` sees the killed PID is gone and reclaims immediately, so steady state is one copy rather than hundreds.

Tests cover removal on drop and on an early return, unique per-process naming, the sweep's pattern and age gates against foreign and look-alike neighbours, immediate reclamation of a dead owner while an in-flight probe is spared, and that the sweep never follows a symlink.

Ships in the next release. Thanks for the clean repro and for separating this from #300.
